content/docs/user-guide/project-structure/dvcyaml-files.md

```diff
@@ -745,6 +752,88 @@ Both individual foreach stages (`train@1`) and groups of foreach stages
 
 </admon>
 
+## `matrix` stages
+
+`matrix` allows you do to define multiple stages based on combinations of
+variables. A `matrix` element accepts one or more variables, each iterating over
+a list of values. For example:
+
+```yaml
+stages:
+  train:
+    matrix:
+      model: [cnn, xgb]
+      feature: [feature1, feature2, feature3]
+    cmd: ./train.py --feature ${item.feature} ${item.model}
+    outs:
+      - ${item.model}.pkl
+```
+
+You can reference each variable in your stage definition using the `item`
+dictionary key. In the above example, you can access `item.model` and
+`item.feature`.
+
+On `dvc repro`, dvc will expand the definition to multiple stages for each
+possible combination of the variables. In the above example, dvc will create six
+stages, one for each combination of `model` and`feature`. The name of the stages
+will be generated by appending values of the variables to the stage name after a
+`@` as with [foreach](#foreach). For example, dvc will create the following
+stages:
+
+- _train@cnn-feature1_
+- _train@cnn-feature2_
+- _train@cnn-feature3_
+- _train@xgb-feature1_
+- _train@xgb-feature2_
+- _train@xgb-feature3_
```

Comment:
Can we change to a code block like this just to save space? The bullet points take up a lot space.

```cli
$ dvc stage list
train@cnn-feature1  Outputs cnn.pkl
train@cnn-feature2  Outputs cnn.pkl
train@cnn-feature3  Outputs cnn.pkl
train@xgb-feature1  Outputs xgb.pkl
train@xgb-feature2  Outputs xgb.pkl
train@xgb-feature3  Outputs xgb.pkl
```

<img width="700" alt="Screenshot 2023-08-16 at 3 36 43 PM" src="https://github.com/iterative/dvc.org/assets/2308172/a5515602-8e0e-4b19-b921-68b609b0e84e">